cloud dm: add start position for incremental data migration (#13704)
pingcap/docs: tidb-cloud/migrate-from-mysql-using-data-migration.md

### Migrate existing data and incremental data

To migrate data to TiDB Cloud once and for all, choose both **Existing data migration** and **Incremental data migration**, which ensures data consistency between the source and target databases.

### Migrate only existing data

To migrate only the existing data of the source database to TiDB Cloud, choose **Existing data migration**.

### Migrate only incremental data

To migrate only the incrementable data of the source database to TiDB Cloud, choose **Incremental data migration**. In this case, the migration job does not migrate the existing data of the source database to TiDB Cloud, but only migrates ongoing changes of the source database to TiDB Cloud.

In the **Start Position** area, you can specify the following types of start positions for incremental data migration:

- The time when the incremental migration job starts
- GTID
- Binlog file name and position

If you do not specify a start position, the migration job will start from the time when the incremental migration job starts.

- **The time when the incremental migration job starts**

>     If you select this option, the migration job will start from the time when the incremental migration job starts to migrate ongoing changes of the source database to TiDB Cloud.

Reviewer: ```suggestion
    If you select this option, the migration job will only replicate the incremental data that generated in the source database after the job starts to TiDB Cloud.
```

- **Specify GTID**

>     Select this option to specify the GTID of the source database, for example, `3E11FA47-71CA-11E1-9E33-C80AA9429562:23`. The migration job will start from the specified GTID to migrate ongoing changes of the source database to TiDB Cloud.

Reviewer 2: `GtidSet` is often an interval. Besides, do we need to revise `start from the specified GTID` to `replicate the events exclude the given gtid set`. /cc @Frank945946
```suggestion
    Select this option to specify the GTID of the source database, for example, `3E11FA47-71CA-11E1-9E33-C80AA9429562:1-23`. The migration job will start from the specified GTID to migrate ongoing changes of the source database to TiDB Cloud.
```

    You can run the following command to check the GTID of the source database:

    ```sql
    SHOW MASTER STATUS;
    ```

    > **Note:**
    >

>     > - Make sure that the GTID is enabled in the source database.

Reviewer: we have to tell the user how to  enbale the  GTID for different type of databases, such as Aurora,GCP MyQL and self hosted mysql. You can refer to the doc
 https://pingcap.feishu.cn/docx/PDgAdbFDEouI8pxPyLhcNxcJnNf

    > - If the source database is MySQL, the MySQL version must be 5.6 or later, and the storage engine must be InnoDB.
    > - If the migration job connects to a secondary database in upstream, the `REPLICATE CREATE TABLE ... SELECT` events cannot be migrated, because the statement will be split into two transactions (`CREATE TABLE` and `INSERT`) that are assigned the same GTID, which will cause the `INSERT` statement to be ignored by the secondary database.

- **Specify binlog file name and position**

>     Select this option to specify the binlog file name (for example, `binlog.000001`) and binlog position (for example, `1307`) of the source database. The migration job will start from the specified binlog file name and position to migrate ongoing changes of the source database to TiDB Cloud.

Reviewer: ```suggestion
    Select this option to specify the binlog file name (for example, `binlog.000001`) and binlog position (for example, `1307`) of the source database. The migration job will start after the specified binlog file name and position to migrate ongoing changes of the source database to TiDB Cloud.
```

    You can run the following command to check the binlog file name and position of the source database:

    ```sql
    SHOW MASTER STATUS;
    ```

>     If there is data in the target database, make sure the binlog position is correct. Otherwise, there might be conflicts between the existing data and the incremental data. If conflicts occur, you need to delete the target database and create a new incremental migration job with the correct binlog position.

Reviewer: There are some other limitations need to be added ,please refer to the  doc https://pingcap.feishu.cn/docx/PDgAdbFDEouI8pxPyLhcNxcJnNf

    - If you use Public IP or VPC Peering, you need to add the Data Migration service's IP addresses to the IP Access List of your source database and firewall (if any).
    - If you use AWS Private Link, you are prompted to accept the endpoint request. Go to the [AWS VPC console](https://us-west-2.console.aws.amazon.com/vpc/home), and click **Endpoint services** to accept the endpoint request.

## Step 3: Choose migration job type

In the **Choose the objects to be migrated** step, you can choose existing data migration, incremental data migration, or both.

### Migrate existing data and incremental data

To migrate data to TiDB Cloud once and for all, choose both **Existing data migration** and **Incremental data migration**, which ensures data consistency between the source and target databases.

### Migrate only existing data

To migrate only existing data of the source database to TiDB Cloud, choose **Existing data migration**.

### Migrate only incremental data

> To migrate only incrementable data of the source database to TiDB Cloud, choose **Incremental data migration**. In this case, the migration job does not migrate existing data of the source database to TiDB Cloud, but only migrates ongoing changes of the source database to TiDB Cloud.

Reviewer 3: ```suggestion
To migrate only the incremental data of the source database to TiDB Cloud, choose **Incremental data migration**. In this case, the migration job does not migrate the existing data of the source database to TiDB Cloud, but only migrates the ongoing changes of the source database that are explicitly specified to TiDB Cloud.
```
